spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java

```diff
@@ -69,6 +74,10 @@
 	private Integer requestTimeout;
 
 	public Map<String, RemoteRepository> getRemoteRepositories() {
+		// Since the remote repositories is a map, we need to remove the default if user provides explicit repo.
+		if (remoteRepositories.size() > 1 && remoteRepositories.containsKey(DEFAULT_REMOTE_REPO_KEY)) {
```

Comment:
yeah, I was hoping to get the map replacing the existing on via the `setter`. But, it turns out that the map just gets `updated` instead of `replaced`. Hence, I went ahead with the above approach. What do you think?